floris/tools/visualization.py

```diff
@@ -71,27 +73,40 @@ def plot_turbines(
         ax.plot([x_0, x_1], [y_0, y_1], color=color)
 
 
-def plot_turbines_with_fi(fi: FlorisInterface, ax=None, color=None, yaw_angles=None):
+def plot_turbines_with_fi(
+    fi: FlorisInterface,
+    ax=None,
+    color=None,
+    wd=None,
+    yaw_angles=None,
+):
     """
     Wrapper function to plot turbines which extracts the data
     from a FLORIS interface object
 
     Args:
-        fi (:py:class:`floris.tools.flow_data.FlowData`):
-                FlowData object.
-        ax (:py:class:`matplotlib.pyplot.axes`): figure axes.
-        color (str, optional): Color to plot turbines
+        fi (:py:class:`floris.tools.floris_interface.FlorisInterface`): FlorisInterface object.
+        ax (:py:class:`matplotlib.pyplot.axes`): Figure axes. Defaults to None.
+        color (str, optional): Color to plot turbines. Defaults to None.
+        wd (list, optional): The wind direction to plot the turbines relative to. Defaults to None.
+        yaw_angles (NDArray, optional): The yaw angles for the turbines. Defaults to None.
     """
     if not ax:
         fig, ax = plt.subplots()
     if yaw_angles is None:
         yaw_angles = fi.floris.farm.yaw_angles
+    if wd is None:
+        wd = fi.floris.flow_field.wind_directions[0]
+
+    # Rotate yaw angles to inertial frame for plotting turbines relative to wind direction
+    yaw_angles = yaw_angles - wind_delta(np.array(wd))
+
     plot_turbines(
         ax,
         fi.layout_x,
         fi.layout_y,
-        yaw_angles[0, 0],
-        fi.floris.farm.rotor_diameters[0, 0],
+        yaw_angles.flatten(),
```

Comment:
Whats the shape of `yaw_angles` here? Previously, it was assumed that it was 3D (wind direction, wind speed, N turbines). Here, it looks like it assumed this is 1D (N turbines). Same question for `fi.floris.farm.rotor_diameters` below.